src/executables/util_sa.c

```diff
@@ -1716,10 +1721,29 @@ diagdb (UTIL_FUNCTION_ARG * arg)
   if (diag == DIAGDUMP_ALL || diag == DIAGDUMP_HEAP)
     {
       bool dump_records;
-      /* this dumps the contents of all heaps */
       dump_records = utility_get_option_bool_value (arg_map, DIAG_DUMP_RECORDS_S);
-      fprintf (outfp, "\n*** DUMP OF ALL HEAPS ***\n");
-      (void) file_tracker_dump_all_heap (thread_p, outfp, dump_records);
+
+      if (class_name == NULL)
+	{
+	  fprintf (outfp, "\n*** DUMP OF ALL HEAPS ***\n");
+	  (void) file_tracker_dump_all_heap (thread_p, outfp, dump_records);
+	}
+      else
+	{
+	  if (diag == DIAGDUMP_ALL)
+	    {
+	      goto print_diag_usage;
+	    }
```

Comment:
```suggestion
     assert (diag != DIAGDUMP_ALL); 
```

I think this is checked redundantly